### PR TITLE
G 691 | Neeraj/Sangitha | Adding confirm and on confirm for HIP initiated linking

### PR DESCRIPTION
**Pull request by Nexengineer**

### CODE REVIEW

File: src/main/java/in/projecteka/gateway/GatewayConfiguration.java [hunk 1]
```diff
@@ -6,21 +6,7 @@
...
-import in.projecteka.gateway.clients.HipConsentNotifyServiceClient;
-import in.projecteka.gateway.clients.HipDataFlowServiceClient;
-import in.projecteka.gateway.clients.HiuConsentNotifyServiceClient;
-import in.projecteka.gateway.clients.IdentityProperties;
-import in.projecteka.gateway.clients.IdentityServiceClient;
-import in.projecteka.gateway.clients.LinkConfirmServiceClient;
-import in.projecteka.gateway.clients.LinkInitServiceClient;
-import in.projecteka.gateway.clients.PatientSearchServiceClient;
+import in.projecteka.gateway.clients.*;
 import in.projecteka.gateway.common.DefaultValidatedRequestAction;
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/GatewayConfiguration.java]: No No.

Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/GatewayConfiguration.java]: No No.

Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/GatewayConfiguration.java]: No No.

Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/GatewayConfiguration.java]: No No.

File: src/main/java/in/projecteka/gateway/SecurityConfiguration.java [hunk 2]
```diff
@@ -97,7 +100,8 @@
             PATH_CONSENT_REQUESTS_ON_INIT,
             PATH_PATIENTS_ON_FIND,
             PATH_HEALTH_INFORMATION_HIP_REQUEST,
-            PATH_HEALTH_INFORMATION_CM_ON_REQUEST
+            PATH_HEALTH_INFORMATION_CM_ON_REQUEST,
+            USERS_AUTH_ON_CONFIRM
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/SecurityConfiguration.java]: I'm just wondering how it's HIU\_HIP API? Isn't it cm supposed to call this?

Comment (Nexengineer) [reply src/main/java/in/projecteka/gateway/SecurityConfiguration.java]: Not sure about this

File: src/main/java/in/projecteka/gateway/user/UserController.java [hunk 3]
```diff
@@ -13,16 +14,14 @@
 import org.springframework.web.bind.annotation.RestController;
 import reactor.core.publisher.Mono;
 
-import static in.projecteka.gateway.common.Constants.PATH_PATIENTS_FIND;
-import static in.projecteka.gateway.common.Constants.PATH_PATIENTS_ON_FIND;
-import static in.projecteka.gateway.common.Constants.X_CM_ID;
-import static in.projecteka.gateway.common.Constants.X_HIU_ID;
+import static in.projecteka.gateway.common.Constants.*;
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/user/UserController.java]: Same No.

Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/user/UserController.java]: Same No.

Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/user/UserController.java]: Same No.

File: src/main/java/in/projecteka/gateway/GatewayConfiguration.java (hunk 1 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/gateway/GatewayConfiguration.java]: :D I remove and this comes back again !!! fixing it 😂

File: src/main/java/in/projecteka/gateway/user/UserController.java [hunk 4]
```diff
@@ -41,4 +40,22 @@
         return patientSearchResponseOrchestrator.processResponse(requestEntity, X_HIU_ID)
                 .subscriberContext(context -> context.put("apiCalled", PATH_PATIENTS_ON_FIND));
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/user/UserController.java:50]:
```suggestion
                .subscriberContext(context -> context.put(API_CALLED, PATH_PATIENTS_ON_FIND));
```

File: src/main/java/in/projecteka/gateway/user/UserController.java [hunk 5]
```diff
@@ -41,4 +40,22 @@
...
+                .map(Caller::getClientId)
+                .flatMap(clientId -> authConfirmRequestOrchestrator
+                        .handleThis(requestEntity, X_CM_ID, X_HIP_ID, clientId)
+                        .subscriberContext(context -> context.put(API_CALLED, USERS_AUTH_CONFIRM)));
+    }
+
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @PostMapping(USERS_AUTH_ON_CONFIRM)
+    public Mono<Void> authOnConfirm(HttpEntity<String> requestEntity){
+        return Mono.empty();
```
Comment (ganesanarun) [inline src/main/java/in/projecteka/gateway/user/UserController.java]: will there be implementation coming?

Comment (Nexengineer) [reply src/main/java/in/projecteka/gateway/user/UserController.java]: yeah from here apporva story kicks in

Comment (Apoorva-GA) [reply src/main/java/in/projecteka/gateway/user/UserController.java]: @Nexengineer We are working on `/links/link/add-contexts` and `/links/link/on-add-contexts` apis on gateway. I don't think it has anything to do with onConfirm.

Comment (Nexengineer) [reply src/main/java/in/projecteka/gateway/user/UserController.java]: On this "on-confirm" you will get the token. Didn't knew where to keep it so stopped here

Comment (Apoorva-GA) [reply src/main/java/in/projecteka/gateway/user/UserController.java]: We had discussed this during kick off right. Saying you will have to implement(Log request)on-confirm on hip  and gateway should call that. That way we can test the flow as well.

Comment (Nexengineer) [reply src/main/java/in/projecteka/gateway/user/UserController.java]: done

File: src/main/java/in/projecteka/gateway/user/UserController.java (hunk 4 above)
Comment (Nexengineer) [reply src/main/java/in/projecteka/gateway/user/UserController.java:50]: done

**Review by Apoorva-GA — APPROVED** [review]

**Pull request merged**